assets/js/components/Work/Tabs/FileSetModal.jsx

```diff
@@ -0,0 +1,309 @@
+import UIFormInput from "@js/components/UI/Form/Input.jsx";
+import UIFormField from "@js/components/UI/Form/Field.jsx";
+import UIFormSelect from "@js/components/UI/Form/Select.jsx";
+import { FILE_SET_ROLES } from "@js/services/global-vars";
+import UISkeleton from "@js/components/UI/Skeleton";
+
+/** @jsx jsx */
+import { css, jsx } from "@emotion/core";
+const dropZone = css`
+  background: #efefef;
+  border: 3px dashed #ccc;
+`;
+
+function FileSetModal({ closeModal, isHidden, workId }) {
+  const [currentFile, setCurrentFile] = useState();
+  const methods = useForm();
+  const [values, setValues] = useState({
+    accession_number: "",
+    description: "",
+    label: "",
+    role: "AM",
+  });
+
+  const { isSubmitting } = methods.formState;
+
+  const { loading: urlLoading, error: urlError, data: urlData } = useQuery(
+    GET_PRESIGNED_URL,
+    {
+      variables: { uploadType: "FILE_SET" },
+      fetchPolicy: "no-cache",
+    }
+  );
+
+  const [ingestFileSet, { loading, error, data }] = useMutation(
+    INGEST_FILE_SET,
+    {
+      onCompleted({ ingestFileSet }) {
+        console.log(ingestFileSet);
+        toastWrapper(
+          "is-success",
+          `FileSet record id: ${ingestFileSet.id} created successfully and ${ingestFileSet.metadata.original_filename} was submitted to the ingest pipeline.`
+        );
+        handleFormReset();
+      },
+      onError({ error }) {
+        console.log(`GraphQL error: ${error}`);
+      },
+      refetchQueries(mutationResult) {
+        return [
+          {
+            query: GET_WORK,
+            variables: { id: workId },
+          },
+        ];
+      },
+    }
+  );
+
+  const handleFormReset = () => {
+    methods.reset();
+    setValues({
+      accession_number: "",
+      description: "",
+      label: "",
+      role: "AM",
+    });
+    setCurrentFile(null);
+    closeModal();
+  };
+
+  const onSubmit = async (data) => {
+    const requestOptions = {
+      method: "PUT",
+      headers: { "Content-Type": "multipart/form-data" },
+      body: currentFile,
+    };
+
+    fetch(urlData.presignedUrl.url, requestOptions)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error("Network response was not ok");
+        }
+        return response.url;
+      })
+      .then((url) => {
+        ingestFileSet({
+          variables: {
+            accession_number: data.accession_number,
+            workId,
+            role: data.role,
+            metadata: {
+              description: data.description,
+              label: data.label,
+              original_filename: currentFile.name,
+              location: s3Location(url),
+            },
+          },
+        });
+      })
+      .catch((error) => {
+        console.error("There was a problem with the S3 upload", error);
+        toastWrapper(
+          "is-danger",
+          `There was a problem with the S3 upload: ${error}`
+        );
+      });
```

Comment:
I'm curious here how an error in the `ingestFileSet` mutation would be handled.  Does the `fetch .catch()` block catch it?   Does it pass through to the mutation error handler?   

If it works, it works, but if there was any overlap in error handling that's a head scratcher, another option may be to use `async/await` to synchronously grab the presigned url, then call `ingestFileSet` and let Apollo client handle any errors.

Reply:
If you're running the branch locally you can cause an error in the `ingestFileSet` mutation by trying to submit the form using an accession_number that already exists. The fetch `catch` block does not catch it. The `onError({ error })` in the mutation recognizes it  - though (and this was a head scratcher for me - the `error` object on line 57 is undefined but the error details are present in the `error` object and printed on line 166/168.  

I can try to rework it to wait for the presigned url on submit like you said.